src/app/onboarding/roles/page.tsx

```diff
@@ -1,57 +1,130 @@
+          code: 'custom',
+          message: 'Must include at least one role',
+        });
+      return ctx;
+    }),
+});
+
 export default function Page() {
   const onboarding = useContext(OnboardingContext);
+  const { push } = useRouter();
 
-  const setRoles = (roles: Set<RoleEnum>) => {
-    onboarding?.setRoles(roles);
-    if (roles.size !== 0) {
-      onboarding?.setCanContinue(true);
-    } else {
-      onboarding?.setProgress(0);
-      onboarding?.setCanContinue(false);
-    }
+  const onSubmit = (values: z.infer<typeof roleSchema>) => {
+    if (!onboarding) throw new Error('Fatal: no onboarding layout detected');
 
-    if (roles.has('ATTORNEY')) {
-      onboarding?.setFlow([
-        { name: 'Roles', url: 'roles' },
-        { name: 'Basic Info', url: 'basic-information' },
-        { name: 'Availability', url: 'availability' },
-        { name: 'Legal Experience', url: 'legal-experience' },
-        { name: 'Done', url: 'done' },
-      ]);
+    const roles = values.roles.split(',') as RoleEnum[];
+    onboarding.setRoles(roles);
+
+    let newFlow: FlowData[];
+
+    if (roles.includes('ATTORNEY')) {
+      newFlow = ATTORNEY_FLOW;
+    } else if (roles.includes('LEGAL_FELLOW')) {
+      newFlow = LEGAL_FELLOW_FLOW;
     } else {
-      onboarding?.setFlow([
-        { name: 'Roles', url: 'roles' },
-        { name: 'Basic Info', url: 'basic-information' },
-        { name: 'Availability', url: 'availability' },
-        { name: 'Done', url: 'done' },
-      ]);
+      newFlow = INTERPRETER_FLOW;
     }
+
+    onboarding.setFlow(newFlow);
+    push(`/onboarding/${newFlow[1].url}`);
   };
 
+  const form = useForm<z.infer<typeof roleSchema>>({
+    resolver: zodResolver(roleSchema),
+    defaultValues: {
+      roles: (onboarding
+        ? Array.from(onboarding.roles.values() || []).join(',')
+        : '') as RoleOptionType,
+    },
+  });
+
   return (
-    <>
-      <H1>Roles</H1>
-
-      <InputDropdown
-        label="What role(s) would you like to hold?"
-        multi
-        defaultValue={onboarding?.roles}
-        onChange={v => setRoles(v as Set<RoleEnum>)}
-        options={roleOptions}
-      />
-    </>
+    <FormProvider {...form}>
+      {/* div to fill top space, delete after nav bar is added */}
+      <div />
+      <SmallCardForm onSubmit={form.handleSubmit(onSubmit)}>
+        <H1>Role</H1>
+
+        <FormField
+          control={form.control}
+          name="roles"
+          render={({ field, fieldState }) => (
+            <FormItem>
+              <FormLabel>What role(s) would you like to hold?</FormLabel>
+              <FormControl>
+                <InputDropdown
+                  error={!!fieldState.error}
+                  onChange={v => field.onChange(v ?? '')}
+                  options={roleOptions}
+                  defaultValue={field.value}
+                />
+              </FormControl>
+              <FormMessage />
+            </FormItem>
+          )}
```

Comment:
are we still planning to add the callout/extra description of each role? if so, would that be added in a separate pr?

Reply:
pending design

Reply:
yeah, that would be a separate PR

Reply:
I can add it in this one once designs are out, since I don't want this to take up another sprint.